### PR TITLE
8135 remove intervenor feedback

### PR DESCRIPTION
**Pull request by rosiehsmith**
On the Edit Party Information form, the Petitioner Type field needs to be updated to match the “Role type” on the Add Party form. The current field only has “petitioner” as the role, so when I went to edit an Intervenor, that field mistakenly is populated by petitioner instead of intervenor.

On the Edit Party Information form, “Remove this petitioner” should be updated to “Remove this Party”

After removing intervenor, success message should say “Intervenor Successfully Removed” and the Intervenor/Participant tab should be selected (unless there are no more intervenors or participants on the case)

The title of the Intervenor/Participant tab should be “Intervenor/Participant(s)”

### CODE REVIEW

File: web-client/src/presenter/actions/caseAssociation/removePetitionerAndUpdateCaptionAction.test.js [hunk 1]
```diff
@@ -28,14 +29,20 @@ describe('removePetitionerAndUpdateCaptionAction', () => {
     };
   });
 
-  it('should set the success message of `Petitioner removed.`', async () => {
+  it('should set the success message of `Petitioner successfully removed.` when a petitioner is removed form the case', async () => {
```
Comment (rachelschneiderman) [inline web-client/src/presenter/actions/caseAssociation/removePetitionerAndUpdateCaptionAction.test.js]: Nit: spelling of `from`

File: web-client/src/presenter/actions/caseAssociation/removePetitionerAndUpdateCaptionAction.test.js [hunk 2]
```diff
@@ -44,7 +51,37 @@ describe('removePetitionerAndUpdateCaptionAction', () => {
...
     });
 
-    expect(output.alertSuccess.message).toBe('Petitioner removed.');
+    expect(output.alertSuccess.message).toBe(
+      'Petitioner successfully removed.',
+    );
+    expect(output.tab).toBe('caseInfo');
+  });
+
+  it('should set the success message of `Intervenor successfully emoved.` when an intervenor is removed form the case', async () => {
```
Comment (rachelschneiderman) [inline web-client/src/presenter/actions/caseAssociation/removePetitionerAndUpdateCaptionAction.test.js]: Nit: spelling of `removed` and `from`

File: web-client/src/views/EditPetitionerInformationInternal.jsx [hunk 3]
```diff
@@ -61,25 +61,28 @@ export const EditPetitionerInformationInternal = connect(
...
-            <FormGroup>
-              <label className="usa-label" htmlFor="petitionerType">
-                <span>Petitioner Type</span>
+            <FormGroup errorText={validationErrors?.contact?.contactType}>
+              <label className="usa-label" htmlFor="contactType">
+                <span>Role type</span>
               </label>
               <select
-                aria-describedby="petitioner-type"
+                aria-describedby="role-type"
```
Comment (rachelschneiderman) [inline web-client/src/views/EditPetitionerInformationInternal.jsx]: Is this correct?

File: shared/src/business/entities/EntityConstants.js [hunk 4]
```diff
@@ -739,10 +739,10 @@ const OTHER_TYPES = {
 const CONTACT_TYPES = {
   primary: 'primary',
   secondary: 'secondary',
-  otherFiler: 'otherFilers',
+  otherFiler: 'otherFiler',
```
Comment (rachelschneiderman) [inline shared/src/business/entities/EntityConstants.js]: Is there a migration for this change? Should we remove `otherFiler` now that it's been replaced by intervenor and participant entries?

File: web-client/pa11y/pa11y-docketclerk.js [hunk 5]
```diff
@@ -227,6 +227,7 @@ module.exports = [
       'wait for #remove-petitioner-modal to be visible',
     ],
     notes: 'checks a11y of remove petitioner confirm modal',
+    only: true,
```
Comment (rachaelparris) [inline web-client/pa11y/pa11y-docketclerk.js]: Oopsie!

File: shared/src/business/entities/contacts/Petitioner.js [hunk 6]
```diff
@@ -107,7 +107,7 @@ Petitioner.VALIDATION_RULES = {
     then: joi.optional().allow(null),
   }),
   title: joi.when('contactType', {
-    is: CONTACT_TYPES.otherFiler,
+    is: CONTACT_TYPES.participant || CONTACT_TYPES.intervenor,
```
Comment (mrinsin) [inline shared/src/business/entities/contacts/Petitioner.js:110]: Nice!

**Review by mrinsin — APPROVED** [review]

**Pull request merged**